### PR TITLE
Photon: Add webp support

### PR DESCRIPTION
**Pull request by kraftbj**
Supports #19654

#### Changes proposed in this Pull Request:

* Adds webp support to Photon

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Using WP 5.8, upload a webp image to your site (see test images at https://developers.google.com/speed/webp/gallery1 )
* Ensure the image CDN feature is active.
* Before the patch, see that Photon URLs are not used for the webp image.
* After patch, see that they are.

### CODE REVIEW

**Review by samiff — COMMENTED** [review]
Tested well for me, was able to see working Photon urls in a test gallery and slideshow using webp images.

@kraftbj I did a quick search to see if there was anything else that might need updating, any of these stand out to you?

https://github.com/Automattic/jetpack/blob/8cde890faf5574692abce73364aefef6aeccb3ea/projects/plugins/jetpack/sal/class.json-api-post-base.php#L583

https://github.com/Automattic/jetpack/blob/74d0505819b31d77294c36bc5b10acb60410c891/projects/plugins/jetpack/class.json-api-endpoints.php#L1370

https://github.com/Automattic/jetpack/blob/ba2dd6c30be137baad96aafb2466d0aed7e182e5/projects/plugins/jetpack/_inc/lib/core-api/class.jetpack-core-api-site-endpoints.php#L224

https://github.com/Automattic/jetpack/blob/7e181422fe121a341647c18de71aa4a802527713/projects/plugins/jetpack/functions.global.php#L381

https://github.com/Automattic/jetpack/blob/ba2dd6c30be137baad96aafb2466d0aed7e182e5/projects/plugins/jetpack/modules/videopress/js/videopress-plupload.js#L247

**Comment by kraftbj** [comment]
@samiff I don't think any of those are required for *Photon* to function, but it does look like they all do to be updated. I would feel more comfortable not including them as part of 10.0, though. With the scope creeping a bit, let's move this to 10.1.

File: projects/plugins/jetpack/tests/php/test_class.jetpack_photon.php
Comment (anomiex) [inline projects/plugins/jetpack/tests/php/test_class.jetpack_photon.php:1487]: Completely optional, but you could avoid defining an extra class by using reflection instead:
```php
$reflectionClass  = new ReflectionClass( Jetpack_Photon::class );
$reflectionMethod = $reflectionClass->getMethod( 'validate_image_url' );
$reflectionMethod->setAccessible( true );
$this->assertEquals( $expected, $reflectionMethod->invoke( null, $url ) );
```

Comment (kraftbj) [reply projects/plugins/jetpack/tests/php/test_class.jetpack_photon.php]: Good idea! 744c14b

**Review by sdixon194 — DISMISSED** [review]
This works for me 👍

**Comment by sdixon194** [comment]
@kraftbj  are you still working on this one, or need approval again after the merging with master?

**Review by sdixon194 — APPROVED** [review]

**Pull request merged**
